feature/dynamodb/enhancedclient/schema_builder_methods.go

```diff
@@ -0,0 +1,250 @@
+package enhancedclient
+
+import (
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+)
+
+func (s *Schema[T]) AttributeDefinitions() []types.AttributeDefinition {
+	return s.attributeDefinitions
+}
+
+func (s *Schema[T]) WithAttributeDefinitions(attributeDefinitions []types.AttributeDefinition) *Schema[T] {
+	s.attributeDefinitions = attributeDefinitions
+
+	return s
+}
+
+func (s *Schema[T]) KeySchema() []types.KeySchemaElement {
+	return s.keySchema
+}
+
+func (s *Schema[T]) WithKeySchema(keySchema []types.KeySchemaElement) *Schema[T] {
+	s.keySchema = keySchema
+
+	return s
+}
+
+func (s *Schema[T]) TableName() *string {
+	return s.tableName
+}
+
+func (s *Schema[T]) WithTableName(tableName *string) *Schema[T] {
```

Comment:
This and other apis where the value is set directly by the dev should be handled as a `string` rather than `*string`